frontend/javascripts/test/sagas/volumetracing/volumetracing_saga_integration.spec.js

```diff
@@ -556,3 +705,164 @@ test.serial("Brushing/Tracing with undo (II)", async t => {
   t.is(await t.context.api.data.getDataValue(volumeTracingLayerName, [1, 0, 0]), newCellId + 1);
   t.is(await t.context.api.data.getDataValue(volumeTracingLayerName, [5, 0, 0]), oldCellId);
 });
+
+test.serial("Brushing/Tracing with upsampling to unloaded data", async t => {
+  const oldCellId = 11;
+  t.context.mocks.Request.sendJSONReceiveArraybufferWithHeaders = createBucketResponseFunction(
+    Uint16Array,
+    oldCellId,
+    500,
+  );
+
+  console.log("zoom step:", getRequestLogZoomStep(Store.getState()));
+  Store.dispatch(setZoomStepAction(4));
+  console.log("zoom step:", getRequestLogZoomStep(Store.getState()));
+
+  // Reload buckets which might have already been loaded before swapping the sendJSONReceiveArraybufferWithHeaders
+  // function.
+  await t.context.api.data.reloadAllBuckets();
+  const volumeTracingLayerName = t.context.api.data.getVolumeTracingLayerIds()[0];
+
+  const paintCenter = [0, 0, 0];
+  const brushSize = 16;
+  const newCellId = 2;
+
+  console.log("Set overwriteMode to OVERWRITE_EMPTY");
+  Store.dispatch(updateUserSettingAction("overwriteMode", OverwriteModeEnum.OVERWRITE_EMPTY));
+
+  Store.dispatch(updateUserSettingAction("brushSize", brushSize));
+  Store.dispatch(setPositionAction([0, 0, 0]));
+  Store.dispatch(setToolAction(AnnotationToolEnum.BRUSH));
+
+  Store.dispatch(setActiveCellAction(newCellId));
+  Store.dispatch(startEditingAction(paintCenter, OrthoViews.PLANE_XY));
+  Store.dispatch(addToLayerAction(paintCenter));
+  Store.dispatch(finishEditingAction());
+
+  await t.context.api.tracing.save();
+
+  for (let zoomStep = 0; zoomStep <= 5; zoomStep++) {
+    console.log("request at zoomStep", zoomStep);
+    t.is(
+      await t.context.api.data.getDataValue(volumeTracingLayerName, [0, 0, 0], zoomStep),
+      oldCellId,
+      `Center should still have old value at zoomstep=${zoomStep}`,
+    );
+  }
+});
+
+test.serial("Erasing on mag 4 where mag 1 is unloaded", async t => {
+  const oldCellId = 11;
+  t.context.mocks.Request.sendJSONReceiveArraybufferWithHeaders = createBucketResponseFunction(
+    Uint16Array,
+    oldCellId,
+    500,
+  );
+
+  Store.dispatch(setZoomStepAction(4));
+
+  // Reload buckets which might have already been loaded before swapping the sendJSONReceiveArraybufferWithHeaders
+  // function.
+  await t.context.api.data.reloadAllBuckets();
+  const volumeTracingLayerName = t.context.api.data.getVolumeTracingLayerIds()[0];
+
+  const paintCenter = [0, 0, 0];
+  // This particular brushSize used to trigger a bug. It should not be changed.
+  const brushSize = 263;
+
+  for (let zoomStep = 0; zoomStep <= 5; zoomStep++) {
```

Comment:
The test says "... where mag 1 is unloaded", but don't these lines load the data in all mags, before erasing?